packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts

```diff
@@ -0,0 +1,18 @@
+import { BaseSchemaValidator, TSchemaValidatorResponse } from './types';
+import Ajv from 'ajv';
+
+export class JsonSchemaValidator extends BaseSchemaValidator {
+  name = "json-schema-validator";
+
+  validate(): TSchemaValidatorResponse {
+    const ajv = new Ajv(this.options);
+    const validator = ajv.compile(this.validationLogic);
+
+    if (!validator(this.data)) {
+      return Promise.resolve({ isValid: false, errorMessage: this.errorMessage || 'Bad Request' });
```

Comment:
I guess that should be Promise.reject, in that case `isValid: false` is pretty obvious, just return the message...

Reply:
I somewhat disagree.
validator should return is valid or not. the validation should be on that field.